### PR TITLE
[FSSDK-12510] fix old test issues with mockito versions

### PR DESCRIPTION
**Pull request by jaeopt**
Fix instrumentation tests on Android API 28+

## Problem
- `TransactionTooLargeException` crashed the `android-sdk` test process when 262+ test results were bundled into a single IPC parcel exceeding Android's 1MB limit
- `Mockito cannot mock this class` failures on API 28+ because the default `mockito-android` mock maker uses a hidden/restricted API (`BaseDexClassLoader#addDexPath()`) blocked since API 28
- Root cause of Mockito failures: PR #516 (targetSDK 35 upgrade) downgraded `mockito_ver_sdk_module` from `3.6.28` → `2.28.2`, which is too old to work with `dexmaker-mockito-inline`

## Changes
- Revert `mockito_ver_sdk_module` back to `3.6.28`
- Add `dexmaker-mockito-inline:2.28.3` to all 6 modules — uses JVMTI bytecode rewriting instead of restricted APIs, works on API 28+
- Add Android Test Orchestrator to `android-sdk` module — runs each test class in its own process, preventing parcel overflow as the test suite grows
- Pin `objenesis:2.6` to maintain API 21 compatibility (objenesis 3.x requires API 26+)
- Set `testCoverageEnabled false` in all modules to avoid JVMTI agent conflict between Jacoco and dexmaker

## Test plan
Existing tests will cover the changes.

## Issues
- FSSDK-12510

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -1,5 +1,6 @@
-# Maven local version (when android-sdk built for local maven repo)
-version = 3.5.0-SNAPSHOT
+# Maven version
+# - keep SNAPSHOT for fallback for local build version
+version = 0.0.0-SNAPSHOT
```
[the remark this answers is not in the document]
Comment (jaeopt) [reply gradle.properties:3]: 0.0.0 is intentional to clarify do-not mange SNAPSHOT version, confusing.f

File: .github/workflows/android.yml [hunk 2]
```diff
@@ -81,18 +81,19 @@ jobs:
...
             arch: x86
             target: google_apis
-          - api-level: 29
+          - api-level: 33
             arch: x86_64
-            target: default
+            target: google_apis
+          - api-level: 36
+            arch: x86_64
+            target: google_apis
```
[the remark this answers is not in the document]
Comment (jaeopt) [reply .github/workflows/android.yml:96]: Comments in the actions are clear enough.

**Review by muzahidul-opti — APPROVED** [review]
Mokito major version upgrade, good win.
